src/data/nat/strong_recursion.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import data.nat.basic
+import tactic
+
+/-!
+# Strong recursion
+
+A strong recursion principle based on `fin`.
+The benefit of `(Π (m:fin n), X m)` over `Π (m:ℕ) (h:m < n), X m`
+is that with `fin` the bound on `m` is bundled,
+and this can be used in later proofs and constructions.
+-/
```

Comment:
The linter will complain on your docstringless defs. Also, an example in the file level docstring illustrating how this can be used would be welcome.

Reply:
Fixed and done.